src/Tribe/Attendees.php

```diff
@@ -729,4 +729,66 @@ public function user_can( $generic_cap, $event_id ) {
 
 		return false;
 	}
+
+	/**
+	 * Determines if the current user (or an ID-specified one) is allowed to delete, check-in, and
+	 * undo check-in attendees. Has to be an allowed role *and* have certain caps.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $user_id Optional. The ID of the user whose access we're checking.
+	 * @return boolean
+	 */
+	public function user_can_manage_attendees( $user_id = 0 ) {
+
+		$user_id  = 0 === $user_id ? get_current_user_id() : $user_id;
+		$user_obj = get_user_by( 'ID', $user_id );
+
+		if ( empty( $user_obj ) || ! isset( $user_obj->roles ) ) {
+			return false;
+		}
+
+		/**
+		 * Allows customizing the roles a user can be while allowed to manage attendees.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $default_roles The roles a user can be while allowed to manage attendees.
+		 */
+		$allowed_roles = apply_filters( 'tribe_tickets_roles_can_manage_attendees', array(
+			'administrator',
+			'editor',
+		) );
+
+		/**
+		 * Allows customizing the caps a user must have to be allowed to manage attendees.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $default_caps The caps a user must have to be allowed to manage attendees.
+		 */
+		$required_caps = apply_filters( 'tribe_tickets_caps_can_manage_attendees', array(
+			'edit_posts',
+			'edit_others_posts',
+		) );
+
+		$allowed = false;
+
+		// First make sure the user is of an allowed role.
+		foreach ( $allowed_roles as $role ) {
+			if ( in_array( $role, $user_obj->roles ) ) {
+				$allowed = true;
+			}
+		}
+
+		// Next make sure the user has proper caps in their role.
+		foreach ( $required_caps as $cap ) {
+			if ( ! user_can( $user_obj, $cap ) ) {
+				$allowed = false;
```

Comment:
Hmm ... seems like dropping the role check would simplify things and that testing for at least _one_ expected capability (instead of _all_ required capabilities) would be more flexible, since it would then be much easier, as an example, for someone to apply a new capability like `process_tickets` to a lower level team member for whom they don't also want to grant post editing powers.

What are your thoughts? That would essentially let us maintain backwards compatibility but open up easier ways for customers to use this to their advantage.

Reply:
@barryhughes I'd personally be okay with reducing this to just looking for, for example, the `edit_others_posts` cap — would you like me to do that, and ditch the explicit role-check as well?

Reply:
Yeah I think that's a good way to go!